src/common/mclock_common.h

```diff
@@ -0,0 +1,186 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2025 IBM, Red Hat
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+
+#pragma once
+#include "config.h"
+#include "ceph_context.h"
+#include "dmclock/src/dmclock_server.h"
+#ifndef WITH_CRIMSON
+ #include "mon/MonClient.h"
+#else
+ #include "crimson/mon/MonClient.h"
+#endif
+
+// scheduler class for classic
+enum class op_scheduler_class : uint8_t {
+  background_recovery = 0,
+  background_best_effort,
+  immediate,
+  client,
+  COUNT,    //Always last
```

Comment:
Couple of things to consider for the scheduler class enums:

1. Is `COUNT` required now that `internal_client_count` is not based on it? Do you see it being useful in any other situation? Same comment applies to `scheduler_class_t` below.

2. Can the two enum classes be merged into one that both classic and crimson can use?

Reply:
We can use a single enum if we are 100% sure we are not using implicit integer value in code anywhere specially during encode/decode of scheduler_class, i am not sure 100% about it that;'s why i used same enum values.

Reply:
I don't think op_scheduler_class is encoded/decoded. The scheduler class is set only after the OpSchedulerItem is created based on the op type. The op type is already decoded at the point of OpSchedulerItem creation.

Reply:
Understood, this can be done when the crimson PR is worked upon. For now, the COUNT enum can probably be removed.